common/src/main/java/io/netty/util/ThreadDeathWatcher.java

```diff
@@ -121,10 +125,10 @@ public static boolean awaitInactivity(long timeout, TimeUnit unit) throws Interr
             throw new NullPointerException("unit");
         }
 
-        Thread watcherThread = ThreadDeathWatcher.watcherThread;
-        if (watcherThread != null) {
-            watcherThread.join(unit.toMillis(timeout));
-            return !watcherThread.isAlive();
+        Thread t = watcherThread.get();
+        if (t != null) {
```

Comment:
Legitime in the sense that it is not forbidden in terms of the API documentation. I don't think it makes a lot of sense tho.. I just would hate to break some app because of such a change and I think we could easily just add a new method that will exactly do this and add this behaviour.

Something like `awaitInactivityAndClose()`